### PR TITLE
Authentication, redirects - #5 Implementing user profile page

### PR DESCRIPTION
**Pull request by djavorek**

### CODE REVIEW

File: client/src/hooks.ts [hunk 1]
```diff
@@ -35,8 +35,9 @@ export async function handle({
 export function getSession(event: RequestEvent) {
 	return event.locals.user
 		? {
-				// Only publish public data here.
+				// Only publish public data here. Do not mix it up with event.locals.user
```
Comment (szattila98) [inline client/src/hooks.ts:38]: What does this mean exactly? What should I not mix up?

File: client/package.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
 		"@smui/textfield": "^6.0.0-beta.14",
 		"@sveltejs/adapter-auto": "next",
 		"@sveltejs/kit": "next",
+		"@types/cookie": "^0.4.1",
```
Comment (szattila98) [inline client/package.json:24]: Why is there a different version in dev dependencies?

File: client/src/routes/index.svelte [hunk 3]
```diff
@@ -1,8 +1,33 @@
+<script context="module">
+	/** @type {import('@sveltejs/kit').Load<{
+	 * 		session: App.Session
+	 * }>} */
+	export async function load({ session }) {
```
Comment (szattila98) [inline client/src/routes/index.svelte:5]: Does this type comment annotation thingy work for you? 
I still have type error: "Binding element 'session' implicitly has an 'any' type"

Comment (szattila98) [reply client/src/routes/index.svelte:5]: It occured to me if I specify the lang attribute in `<script>` it might work and seems like it does, then import the types necessary. I also used an arrow function for `load` with the `Load` type..
image

Comment (djavorek) [reply client/src/routes/index.svelte:5]: Good point, it was just copy-pasted from the docs, but I was fed up with it.
Will try your solution.

File: client/src/hooks.ts (hunk 1 above)
Comment (djavorek) [reply client/src/hooks.ts:38]: It just means, that probably this will be just a subset of all the available data in _event.locals.user_.
This is publicly available (on the session), while  _event.locals.user_ will contain all data from the user.

I will add reword this to be more concise

File: client/src/routes/index.svelte (hunk 3 above)
Comment (djavorek) [reply client/src/routes/index.svelte]: Works much better now.

**Review by szattila98 — APPROVED** [review]
